### ISSUE TITLE
Cannot get eslint to work

### ISSUE THREAD

**Issue by kevinSuttle** [eslint/eslint#2440]
No matter what I try—global, local, different config files and formats—doesn't matter. I can't get eslint to work.

```
                    throw new Error("Definition for rule '" + key + "' was not
                          ^
Error: Definition for rule 'space-unary-word-ops' was not found.
```

**.eslintrc**
```
env:
  # Browser global variables.
  browser: true
  # Configure for Node.js environment
  node: true

# globals:
  #########################
  ## Only add globals if you're absolutely certain they need to be globals
  ##########################
  # console: true

#########################
## set to 0 to allow
## set to 1 to disallow as warning
## set to 2 to disallow as error
#########################
rules:
  #########################
  ## Optional Rules
  #########################
  # Disallow use of `console`
  no-console: 2

  # Disallow warning comments
  no-warning-comments:
    - 1
    - terms
        - todo
        - fixme
      location
        - anywhere

  # Warns when variables are defined but never used
  no-unused-vars: 1

  # Enforces comma style (first or last)
  comma-style:
    - 2
    - last

  # Enforces one true `this` variable
  consistent-this:
    - 2
    - self
  # Allows dangling underscores in identifiers
  no-underscore-dangle: 2

  # Enforces function expressions to have a name
  func-names: 0

  # Set maximum depth of nested callbacks
  max-nested-callbacks:
    - 1
    - 3

  #########################
  ## Core Rules
  ##########################
  # Enforces camel case names
  camelcase: 2

  # Prohibit use of == and != in favor of === and !==
  eqeqeq: 2

  # Suppresses warnings about == null comparisons
  no-eq-null: 2

  # No mixing tabs and spaces, with 2 spaces only
  no-mixed-spaces-and-tabs: 2

  # Prohibits use of a variable before it is defined
  no-use-before-define: 2

  # Requires capitalized names for constructor functions
  new-cap: 2

  # Prohibits use of explicitly undeclared variables
  no-undef: 2

  # Enforces Use Strict at the top of function scope
  strict: 2
  # Prohibits global Use Strict
  global-strict: 2

  # Requires variable declarations to be at the top
  vars-on-top: 2

  # Enforce curly braces around blocks in loops and conditionals
  curly: 2

  # Prohibits the use of immediate function invocations w/o wrapping in parentheses
  wrap-iife: 2

  # Prohibits `argument.caller` and `argument.callee`
  no-caller: 2

  # Requires all `for in` loops to filter object's items
  guard-for-in: 2

  # Prohibits comparing a variable against itself
  no-self-compare: 2

  # Prohibits use of `undefined` variable
  no-undefined: 0

  # Prohibits nested ternaries
  no-nested-ternary: 2

  # Enforces space following unary word operators
  space-unary-word-ops: 2

  # Enforces a space before blocks
  space-before-blocks:
    - 2
    - always

  # Enforces spaces following keywords
  space-after-keywords:
    - 2
    - always
    - checkFunctionKeyword: true

  # Enforces quoted property names
  quote-props:
    - 2
    - always

  # Enforces padded blocks
  padded-blocks:
    - 1
    - never

  # Enforce functions as expressions
  func-style:
    - 2
    - expression

  # Require brace style
  brace-style:
    - 2
    - stroustrup

  # Prohibits Yoda conditions
  yoda:
    - 2
    - never

  # Enforce use of single quotation marks for strings.
  quotes:
    - 2
    - single

  # Enforces space inside of brackets (except property name)
  space-in-brackets:
    - 2
    - always
    - propertyName: false
      singleValue: false

# ESlint Plugins
plugins:
  - eslint-plugin-one-variable-per-var

``` 

OSX 10.10.3 (14D136)
node v0.12.2
npm  v2.9.0

```
❯ which eslint
/usr/local/bin/eslint

❯ eslint -v
v0.20.0

**Comment by hzoo** [comment]
Looks like the `space-unary-word-ops` rule is deprecated.

**Comment by ilyavolodin** [comment]
In your .eslintrc remove this line:
```
space-unary-word-ops: 2
```
This will allow you to run eslint. @hzoo is correct, `space-unary-ops` is a new rule that replaces `space-unary-word-ops`.

**Comment by kevinSuttle** [comment]
Nope. Changed that rule, exact same errors. Removed it entirely. Same output as well.

**Comment by hzoo** [comment]
That's odd. The error is because the rule isn't in eslint anymore - `Error: Definition for rule 'space-unary-word-ops' was not found.` so I would of thought removing it would make it work or at least show a different error..

**Comment by kevinSuttle** [comment]
I uninstalled, removed everything from my package.json and node_modules. Reinstalled. I explicitly pointed with the `-c` flag to my .eslintrc file, which no longer has the rule. Same issue.

**Comment by kevinSuttle** [comment]
Is there an environment variable I can check?

**Comment by kevinSuttle** [comment]
Hm. I just searched the source of this repo, and can't find this line:

`Definition for rule 'space-unary-word-ops' was not found.`

**Comment by kevinSuttle** [comment]
I'm not using NVM btw, so multiple versions seems unlikely.

**Comment by kevinSuttle** [comment]
Full stack trace
```
❯ eslint -c ".eslintrc" app.js
/usr/local/lib/node_modules/eslint/lib/eslint.js:648
                    throw new Error("Definition for rule '" + key + "' was not
                          ^
Error: Definition for rule 'space-unary-word-ops' was not found.
    at /usr/local/lib/node_modules/eslint/lib/eslint.js:648:27
    at Array.forEach (native)
    at EventEmitter.module.exports.api.verify (/usr/local/lib/node_modules/eslint/lib/eslint.js:621:16)
    at processFile (/usr/local/lib/node_modules/eslint/lib/cli-engine.js:197:27)
    at /usr/local/lib/node_modules/eslint/lib/cli-engine.js:299:26
    at walk (/usr/local/lib/node_modules/eslint/lib/util/traverse.js:81:9)
    at /usr/local/lib/node_modules/eslint/lib/util/traverse.js:102:9
    at Array.forEach (native)
    at traverse (/usr/local/lib/node_modules/eslint/lib/util/traverse.js:101:11)
    at CLIEngine.executeOnFiles (/usr/local/lib/node_modules/eslint/lib/cli-engine.js:290:9)
```

**Comment by nzakas** [comment]
Can you run in debug mode?

```
$ DEBUG=eslint:* eslint app.js
```

Let us know what the output is.

**Comment by kevinSuttle** [comment]
AHA! You found it @nzakas. Somehow had the exact same file in my top-level code directory.

```bash
❯ DEBUG=eslint:* eslint app.js
  eslint:cli Running on files +0ms
  eslint:cli-engine Processing app.js +7ms
  eslint:cli-engine Linting /Users/kevinsuttle/Code/REDACTED/app.js +2ms
  eslint:config Constructing config for /Users/kevinsuttle/Code/REDACTED/app.js +0ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Using /Users/kevinsuttle/Code/REDACTED/.eslintrc +10ms
  eslint:config Using /Users/kevinsuttle/Code/.eslintrc +7ms
  eslint:config Merging command line environment settings +2ms
/usr/local/lib/node_modules/eslint/lib/eslint.js:648
                    throw new Error("Definition for rule '" + key + "' was not
                          ^
Error: Definition for rule 'space-unary-word-ops' was not found.
    at /usr/local/lib/node_modules/eslint/lib/eslint.js:648:27
    at Array.forEach (native)
    at EventEmitter.module.exports.api.verify (/usr/local/lib/node_modules/eslint/lib/eslint.js:621:16)
    at processFile (/usr/local/lib/node_modules/eslint/lib/cli-engine.js:197:27)
    at /usr/local/lib/node_modules/eslint/lib/cli-engine.js:299:26
    at walk (/usr/local/lib/node_modules/eslint/lib/util/traverse.js:81:9)
    at /usr/local/lib/node_modules/eslint/lib/util/traverse.js:102:9
    at Array.forEach (native)
    at traverse (/usr/local/lib/node_modules/eslint/lib/util/traverse.js:101:11)
    at CLIEngine.executeOnFiles (/usr/local/lib/node_modules/eslint/lib/cli-engine.js:290:9)
```

**Comment by kevinSuttle** [comment]
Thanks all for your help.

**Comment by nzakas** [comment]
Sure thing!
